app/models/form.rb

```diff
@@ -175,9 +175,10 @@ def visible_questionings
     questionings.reject{|q| q.hidden?}
   end
 
-  # returns questionings that work with sms forms and are not hidden
+  # returns hash of questionings that work with sms forms and are not hidden
   def smsable_questionings
-    questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings = questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings.index_by.with_index { |_, i| i + 1 }
```

Comment:
I appreciate the exchange. I ended up picking up a digital copy of [*Ruby Under a Microscope*](http://patshaughnessy.net/ruby-under-a-microscope) since I'm not satisfied with my understanding of the MRI internals. Looking at the Rubinius implementation of `with_index` is helpful, but not helpful enough. :+1: